### ISSUE TITLE
Add support for the `:kind` command in the REPL

### ISSUE THREAD

**Issue by Gedochao** [scala/scala3#21655]
The `:kind` command was (and is) supported by the Scala 2 REPL, while it is missing in Scala 3.
```scala
scala -S 2.13
Welcome to Scala 2.13.15 (OpenJDK 64-Bit Server VM, Java 17).
Type in expressions for evaluation. Or try :help.

scala> :help
All commands can be abbreviated, e.g., :he instead of :help.
(...)
:kind [-v] <type>        display the kind of a type. see also :help kind
(...)

scala> :kind Int
Int's kind is A

scala> :kind List
List's kind is F[+A]
```

**Comment by aherlihy** [comment]
Scala 2 reference: https://github.com/scala/scala/pull/2340/files

**Comment by SethTisue** [comment]
yeah I think this is little-used, something-someone-randomly-hacked-in-back-in-the-day territory

**Comment by okouyad** [comment]
Is there a particular reason why this was put aside, because I don’t think this was an obscure or of little use ? I am curious as to that was the motivation to not do anything about it ?

I will however prepare a write up to explain why imho it is fundamental to add back this capability to the scala Repl in Scala 3

**Comment by Gedochao** [comment]
> IIs there a particular reason why this was put aside, because I don’t think this was an obscure or of little use ? I am curious as to that was the motivation to not do anything about it ?

@okouyad I believe it simply wasn't much requested by the community. It's never been implemented in the Scala 3 REPL and it wasn't ever requested much. I remember some users did ask for it, but it was never  raised as a high priority missing feature. 

Given that Scala 3 development is largely community driven, and the maintenance team capacity is rather limited, we need carefully pick what needs to be worked on by us.

cc @SethTisue @aherlihy 

> I will however prepare a write up to explain why imho it is fundamental to add back this capability to the scala Repl in Scala 3

All that being said, I am very much interested to hear about your use case. We might reconsider to revisit this, if a good case is presented. And, of course, we always welcome external contributions.
